src/interface/blas1/rotmg.cpp.in

```diff
@@ -72,11 +72,22 @@ template typename SB_Handle::event_t _rotmg(
     BufferIterator<${DATA_TYPE}> _y1, BufferIterator<${DATA_TYPE}> _param,
     const typename SB_Handle::event_t& dependencies);
 
+template typename SB_Handle::event_t _rotmg(
+    SB_Handle& sb_handle, BufferIterator<${DATA_TYPE}> _d1,
+    BufferIterator<${DATA_TYPE}> _d2, BufferIterator<${DATA_TYPE}> _x1,
+    ${DATA_TYPE} _y1, BufferIterator<${DATA_TYPE}> _param,
+    const typename SB_Handle::event_t& dependencies);
+
 #ifdef SB_ENABLE_USM
 template typename SB_Handle::event_t _rotmg(
     SB_Handle& sb_handle, ${DATA_TYPE} * _d1, ${DATA_TYPE} * _d2,
     ${DATA_TYPE} * _x1, ${DATA_TYPE} * _y1, ${DATA_TYPE} * _param,
     const typename SB_Handle::event_t& dependencies);
+
+template typename SB_Handle::event_t _rotmg(
+    SB_Handle& sb_handle, ${DATA_TYPE} * _d1, ${DATA_TYPE} * _d2,
+    ${DATA_TYPE} * _x1, ${DATA_TYPE} _y1, ${DATA_TYPE} * _param,
+    const typename SB_Handle::event_t& dependencies);
```

Comment:
I don't think so. PortBLAS generate the correct interface and `include/interface` and `include/operation` have a templated functions that are compatible with both interfaces. 
I think the interface is already implemented in the library build since tests link to the library and they are not build as header only. You can check looking at [this cmake line](https://github.com/codeplaysoftware/portBLAS/blob/main/test/unittest/CMakeLists.txt#L134) and I paste here the compiler/linker output for `rotmg` test:
```bash
[94/115] : && /opt/intel/oneapi/2024.2/bin/icpx -Wno-deprecated-declarations  -Wno-deprecated-copy-with-user-provided-copy -D__SYCL_DISABLE_NAMESPACE_INLINE__=ON -O3 -Xclang -cl-mad-enable -Wno-error=tautological-constant-compare -mllvm=-loopopt=0 -fsycl -fsycl-targets=spir64 -DSB_ENABLE_USM=1 benchmark/portblas/CMakeFiles/bench_rotmg.dir/blas1/rotmg.cpp.o benchmark/portblas/CMakeFiles/bench_rotmg.dir/main.cpp.o -o benchmark/portblas/bench_rotmg  -Wl,-rpath,/home/nicolo/Documents/git/portBLAS/cpu_build/benchmark/googlebench-build/src:/home/nicolo/Documents/git/portBLAS/cpu_build:  benchmark/googlebench-build/src/libbenchmark.s
o.1.6.1  libportblas.so.0.1.0  benchmark/libbench_info.a  /opt/intel/oneapi/2024.2/lib/libsycl.so  /usr/lib/x86_64-linux-gnu/libopenblas.so ...
```
 and terminal output of `ldd` on `blas1_rotmg_test`:

```bash
ldd blas1_rotmg_test
        linux-vdso.so.1 [...]
        libgtest_main.so.1.12.1 => /home/nicolo/Documents/git/portBLAS/cpu_build/lib/libgtest_main.so.1.12.1 [...]
        libportblas.so.0.1.0 => /home/nicolo/Documents/git/portBLAS/cpu_build/libportblas.so.0.1.0 
       ....
```

Reply:
Right I can see it works for `include/interface`. It wouldn't work for the tree interface since the template parameter is the same for the 4 parameters but I'd say we don't need to expand that interface. Thanks